### ISSUE TITLE
Stuck on "1. Set Up Gateway - Identify Host" on SMC

### ISSUE THREAD

**Issue by cheolminhwang** [OracleSiebel/ConfiguringSiebel#48]
Hi,

I managed to build docker version Siebel 18.9 and open SMC; but even with many combination of hostname:port, getting "Invalid Gateway Host Name and/or HTTPS Port". 
Log says _"[FATAL] 2018-11-30 22:53:14.509 [https-jsse-nio-9011-exec-4] CommonLogger - com.siebel.swsm.util.CGUtil:initializeServiceDiscovery Unable to discover cloud gateway service.Check system configuration, network connection,node availability and zookeeper configuration : unknown protocol: cgwent1.sub09121804010.vcn0912180401.oraclevcn.com
"_
Also tried with script setGateway, setGatewaySecProfile, etc, but no luck so far (PFB output)...

What would be the issue? REST is not working, maybe?

____________________________________________________________________________________
[root@ol7 smc]# ./setGateway -u sadmin -p sadmin -g cgwENT1.sub09121804010.vcn0912180401.oraclevcn.com -h 4430 -a localhost -b 8011
sadmin
sadmin
cgwENT1.sub09121804010.vcn0912180401.oraclevcn.com
4430
localhost
8011

{"CGHostURI":"cgwENT1.sub09121804010.vcn0912180401.oraclevcn.com:4430"}

* About to connect() to localhost port 8011 (#0)
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 8011 (#0)
* Initializing NSS with certpath: sql:/etc/pki/nssdb
* skipping SSL peer certificate verification
* SSL connection using TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
* Server certificate:
*       subject: CN=*.sub09121804010.vcn0912180401.oraclevcn.com
*       start date: Nov 26 17:18:32 2018 GMT
*       expire date: Nov 23 17:18:32 2028 GMT
*       common name: *.sub09121804010.vcn0912180401.oraclevcn.com
*       issuer: CN=ol7
> POST /siebel/v1.0/cginfo HTTP/1.1
> User-Agent: curl/7.29.0
> Host: localhost:8011
> Accept: */*
> Content-Length: 71
> Authorization: Basic c2FkbWluOnNhZG1pbg==
> Content-Type: application/json
>
* upload completely sent off: 71 out of 71 bytes
< HTTP/1.1 401
< Content-Length: 0
< Date: Sat, 01 Dec 2018 00:21:57 GMT
<
* Connection #0 to host localhost left intact
____________________________________________________________________________________________________________
[root@ol7 smc]# ./setGatewaySecProfile -u sadmin -p sadmin -a localhost -b 8011 -d ol7.sub09121804010.vcn0912180401.oraclevcn.com  -e 1521 -f cdb1.9ub09121804010.vcn0912180401.oraclevcn.com -t SIEBEL -v SADMIN -w SADMIN_1

{"Profile":{"ProfileName":"secprof"}, "SecurityConfigParams": {"DataSources": [ {"Name":"secprof", "Type":"DB", "Host":"ol7.sub09121804010.vcn0912180401.oraclevcn.com", "Port":"1521", "SqlStyle":"Oracle", "Endpoint":"cdb1.9ub09121804010.vcn0912180401.oraclevcn.com", "TableOwner":"SIEBEL", "CRC":"", "HashUserPwd":"false"} ], "SecAdptMode":"DB", "SecAdptName":"DBSecAdpt", "DBSecurityAdapterDataSource":"secprof", "DBSecurityAdapterPropagateChange":"false", "NSAdminRole":["Siebel Administrator"], "TestUserName":"SADMIN", "TestUserPwd":"SADMIN_1" }}

* About to connect() to localhost port 8011 (#0)
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 8011 (#0)
* Initializing NSS with certpath: sql:/etc/pki/nssdb
* skipping SSL peer certificate verification
* SSL connection using TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
* Server certificate:
*       subject: CN=*.sub09121804010.vcn0912180401.oraclevcn.com
*       start date: Nov 26 17:18:32 2018 GMT
*       expire date: Nov 23 17:18:32 2028 GMT
*       common name: *.sub09121804010.vcn0912180401.oraclevcn.com
*       issuer: CN=ol7
> POST /siebel/v1.0/cloudgateway/GatewaySecurityProfile HTTP/1.1
> User-Agent: curl/7.29.0
> Host: localhost:8011
> Accept: */*
> Content-Length: 550
> Authorization: Basic c2FkbWluOnNhZG1pbg==
> Content-Type: application/json
>
* upload completely sent off: 550 out of 550 bytes
< HTTP/1.1 500
< Content-Type: text/html;charset=utf-8
< Content-Language: en
< Content-Length: 1096
< Date: Sat, 01 Dec 2018 00:24:11 GMT
< Connection: close
<
<!doctype html><html lang="en"><head><title>HTTP Status 500 – Internal Server Error</title><style type="text/css">h1 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:22px;} h2 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:16px;} h3 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:14px;} body {font-family:Tahoma,Arial,sans-serif;color:black;background-color:white;} b {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;} p {font-family:Tahoma,Arial,sans-serif;background:white;color:black;font-size:12px;} a {color:black;} a.name {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 500 – Internal Server Error</h1><hr class="line" /><p><b>Type</b> Status Report</p><p><b>Message</b> no protocol: /GatewaySecurityProfile</p><p><b>Description</b> The server encountered an unexpected condition that prevented it from fulfilling the re* Closing connection 0
quest.</p><hr class="line" /><h3>Apache Tomcat/8.5.23</h3></body></html>

_______________________________________________________________________________________________
[root@ol7 smc]# docker ps
CONTAINER ID        IMAGE                                     COMMAND             CREATED             STATUS              PORTS                                            NAMES
b6aba9161855        registry.local.com:5000/siebel/ses:18.9   "/bin/bash"         2 hours ago         Up 2 hours                                                           sesENT1
114bae249157        registry.local.com:5000/siebel/cgw:18.9   "/bin/bash"         2 hours ago         Up 2 hours          0.0.0.0:8001->9001/tcp                           cgwENT1
84d563067c44        registry.local.com:5000/siebel/sai:18.9   "/bin/bash"         2 hours ago         Up 2 hours          0.0.0.0:4439->4430/tcp, 0.0.0.0:8011->9011/tcp   saiENT1
_______________________________________________________________________________________________________

Thanks in advance,
Cheolmin Hwang

**Comment by OracleSiebel** [comment]
The most likely issue will be that the SAI cannot ping the CGW, or if it can, that it cannot communicate securely. The address you're using looks OK, but double check it. If that fails, get into that container using docker exec and ensure you can ping the CGW. Likely you can't, in which case restart the containers using -net options to put all the siebel containers in the same virtual network.

**Comment by OracleSiebel** [comment]
Before you attempt config with scripts, you should verify that you can do each step manually using the UI in order to confirm the whole setup is working. So, once SAI and CGW are running, connect using a browser to the expose SAI port for HTTPS and attempt login and gateway config. If you cannot get through step 1 of configuring the gateway, it means that the certificate is probably not correct on both SAI and CGW. At this stage, the issue is not with the build, but with basic setup and MOS should be your route to resolve the issue. This topic may help you get started, which describes how to setup tomcat for debugging. If you follow this, you should be able to get to the actual error in the tomcat log file to help understand why the certificate is not working.

How To Setup Tomcat Debug Options For Troubleshooting Purposes(Doc ID 2371192.1)

https://mosemp.us.oracle.com/epmos/faces/KMConsolidatedSearch?_adf.ctrl-state=n7fdhceoo_16&_afrLoop=274790104728773#

**Comment by cheolminhwang** [comment]
Thanks for the information, I will verify the env, especially certificate related.

**Comment by mohanbayyana** [comment]
i am also getting HTTP 500 error while invoking below URL. Can anybody help us. i am trying to install docker siebel configuration in AWS servers. 

https://$SAIHOST:$SAIPORT/siebel/v1.0/cloudgateway/GatewaySecurityProfile 

About to connect() to localhost port 4443 (#0)
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 4443 (#0)
* Initializing NSS with certpath: sql:/etc/pki/nssdb
* skipping SSL peer certificate verification
* SSL connection using TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
* Server certificate:
*       subject: CN=ip-10-65-35-168.corporate.t-mobile.com
*       start date: Dec 17 22:59:05 2018 GMT
*       expire date: Dec 17 22:59:05 2019 GMT
*       common name: ip-10-65-35-168.corporate.t-mobile.com
*       issuer: CN=ip-10-65-35-168.corporate.t-mobile.com
> POST /siebel/v1.0/cloudgateway/GatewaySecurityProfile HTTP/1.1
> User-Agent: curl/7.29.0
> Host: localhost:4443
> Accept: */*
> Content-Length: 514
> Authorization: Basic c2FkbWluOmFkbWluYWRtaW4=
> Content-Type: application/json
>
* upload completely sent off: 514 out of 514 bytes
< HTTP/1.1 500
< Content-Type: text/html;charset=utf-8
< Content-Language: en
< Content-Length: 1096
< Date: Mon, 07 Jan 2019 18:55:14 GMT
< Connection: close
<
<!doctype html><html lang="en"><head><title>HTTP Status 500 – Internal Server Error</title><style type="text/css">h1 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:22px;} h2 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:16px;} h3 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:14px;} body {font-family:Tahoma,Arial,sans-serif;color:black;background-color:white;} b {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;} p {font-family:Tahoma,Arial,sans-serif;background:white;color:black;font-size:12px;} a {color:black;} a.name {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 500 – Internal Server Error</h1><hr class="line" /><p><b>Type</b> Status Report</p><p><b>Message</b> no protocol: /GatewaySecurityProfile</p><p><b>Description</b> The server encountered an unexpected condition that prevented it from fulfilling the re* Closing connection 0

**Comment by OracleSiebel** [comment]
@cheolminhwang - Were you able to resolve your issues?

**Comment by mohanbayyana** [comment]
> @cheolminhwang - Were you able to resolve your issues?

Any idea what you did to resolve this issue??

**Comment by OracleSiebel** [comment]
@mohanbayyana Have you set up Tomcat for debugging as suggested above in order to verify your certificate is not the issue?

**Comment by mohanbayyana** [comment]
> @mohanbayyana Have you set up Tomcat for debugging as suggested above in order to verify your certificate is not the issue?
that link is not working, can you share valid link?

**Comment by mohanbayyana** [comment]
[ERROR] 2019-01-08 03:57:41.430 [https-jsse-nio-9011-exec-7] CommonLogger - com.siebel.swsm.util.CGUtil:initializeServiceDiscovery Unable to create service discovery instance,Invalid authtoken Or Trust/Key store password.
[ERROR] 2019-01-08 03:57:41.430 [https-jsse-nio-9011-exec-7] CommonLogger - com.siebel.swsm.util.CGUtil:initializeServiceDiscovery Unable to create service discovery instance
[ERROR] 2019-01-08 03:57:41.436 [https-jsse-nio-9011-exec-7] CommonLogger - com.siebel.swsm.config.CloudSourcedConfigReader:getGatewaySecurityProfile Error occurred while retrieving gateway security profile.Check SMC configuration and network connectivity. : ip-10-65-35-168.corporate.t-mobile.com
[FATAL] 2019-01-08 03:57:44.704 [https-jsse-nio-9011-exec-2] CommonLogger - com.siebel.swsm.util.CGUtil:initializeServiceDiscovery Unable to discover cloud gateway service.Check system configuration, network connection,node availability and zookeeper configuration : Illegal base64 character 20
[ERROR] 2019-01-08 03:57:44.706 [https-jsse-nio-9011-exec-2] CommonLogger - com.siebel.integration.eairest.enterprises.EnterprisesResource:connectToExtHTTPSURL Unable to process CG request to external URL : ip-10-65-35-168.***************
[FATAL] 2019-01-08 03:57:47.901 [https-jsse-nio-9011-exec-7] CommonLogger - com.siebel.swsm.util.CGUtil:initializeServiceDiscovery Unable to discover cloud gateway service.Check system configuration, network connection,node availability and zookeeper configuration : Illegal base64 character 20
[ERROR] 2019-01-08 03:57:47.903 [https-jsse-nio-9011-exec-7] CommonLogger - com.siebel.integration.eairest.enterprises.EnterprisesResource:connectToExtHTTPSURL Unable to process CG request to external URL : ip-10-65-35-168.***********
[FATAL] 2019-01-08 03:57:54.090 [https-jsse-nio-9011-exec-2] CommonLogger - com.siebel.swsm.util.CGUtil:initializeServiceDiscovery Unable to discover cloud gateway service.Check system configuration, network connection,node availability and zookeeper configuration : Illegal base64 character 20
[ERROR] 2019-01-08 03:57:54.095 [https-jsse-nio-9011-exec-2] CommonLogger - com.siebel.integration.eairest.enterprises.EnterprisesResource:connectToExtHTTPSURL Unable to process CG request to external URL : ip-10-65-35-168.*********

**Comment by OracleSiebel** [comment]
@mohanbayyana Apologies for the invalid link. That is now corrected. It looks like you found it anyway. The first error seems to indicate a problem with the SSL certificate, thus SAI is unable to talk to the CGW. Both the SAI and the CGW must have the same trust store content. Is the certificate you are using working for a standard install (i.e. non Docker)?. You might want to try going through the SSL generation step here (https://github.com/OracleSiebel/ConfiguringSiebel/tree/master/Containerization/Docker/ol7/manage/SSL) and using *.t-mobile.com instead of a more specific address.

**Comment by cheolminhwang** [comment]
sorry for the late updates; I'm on leave nowadays and will resume this task at the end of January (Jan 2019)

Best Regards,
Cheolmin

**Comment by OracleSiebel** [comment]
@mohanbayyana How are you launching your containers? Are they all launched on the same server. Have you started them in the same docker network?

**Comment by mohanbayyana** [comment]
> @mohanbayyana How are you launching your containers? Are they all launched on the same server? Have you started them in the same docker network?

we had Amazon Linux server, installed docker engine and created images as per your directions. 
created same network under all 3 containers.

**Comment by mohanbayyana** [comment]
If you share with kubernates orchestration, it would be helpful. Thanks

**Comment by cheolminhwang** [comment]
Hello,

Sorry for late updates.
I built new environment and went to 19.01 with current code here; it went fine and Siebel is up and running.
My previous issue should be related with certificate. (Unfortunately I lost my previous free oracle cloud env that I had docker, and I had to rebuild env on my machine; so cannot confirm the issue was surely by certificate though.)

Thanks and Regards,
Cheolmin
